src/main/resources/api.yaml

```diff
@@ -87,7 +87,117 @@ paths:
           description: User is not authorized
         404:
           description: Project not found
-  /projects/{id}/sprints:
+  /projects/{projectId}/members:
+    get:
+      tags:
+        - projects
+      summary: Retrieves users belonging to the project
+      operationId: getUsersFromProject
+      produces:
+        - application/json
+      parameters:
+        - in: path
+          name: projectId
+          description: Project ID
+          required: true
+          type: integer
+          format: int64
+      responses:
+        200:
+          description: Users retrieved successfully
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/ProjectMemberDto'
+        404:
+          description: Project not found
+    post:
+      tags:
+        - projects
+      summary: Adds a user to the project
+      operationId: addUserToProject
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: path
+          name: projectId
+          description: Project ID
+          required: true
+          type: integer
+          format: int64
+        - in: body
+          name: request
+          description: Information identifying the user
+          required: true
+          schema:
+            $ref: '#/definitions/UserIdentificationDto'
+      responses:
+        200:
+          description: User added successfully
+          schema:
+            $ref: '#/definitions/ProjectMemberDto'
+  /projects/{projectId}/members/{userId}:
+    put:
+      tags:
+        - projects
+      summary: Sets user's role in the project
+      operationId: setUserRole
```

Comment:
Do czego to potrzebuje `projectId` + `userId`  + obiektu `ProjectMemberDto`? Nie wystarczyłaby rola zamiast tego `ProjectMemberDto`?

Reply:
Noo w teorii struktura taka sama jaką zwraca GET :thinking: proponujesz zmienić?

Reply:
A no może spoko, jeszcze jutro to przemyślę świeższym mózgiem

Reply:
@jplewa ping

Reply:
A to nie powinien być patch albo put na `/projects/{projectId}/members/{userId}/role`?

Reply:
PUT na `/projects/{projectId}/members/{userId}` powinien wysyłać `ProjectMemberDto`
PUT na `/projects/{projectId}/members/{userId}/role` powinien wysłać samą rolę

W tym momencie oba requesty są równoważne bo jedyna cecha zmienialna to `role`. Różnica pojawi się jak coś się doda do `ProjectMemberDto`. Jak robimy szefowo?

PATCH raczej nie ma sensu bo nie zmieniamy części zasobu tylko jego całość.

Reply:
W sensie nie PATCH na `/projects/{projectId}/members/{userId}/role` tylko na `/projects/{projectId}/members/{userId}`

W tym `ProjectMemberDto` jest sporo informacji, które mi się nie wydają super konieczne tutaj

Reply:
> W sensie nie PATCH na `/projects/{projectId}/members/{userId}/role` tylko na `/projects/{projectId}/members/{userId}`

Tak, tak, chodziło mi o to, że jedyne pole które można zmienić to rola więc de-facto ustawiamy cały zasób.

Zmienię w takim razie na PUT `/projects/{projectId}/members/{userId}/role`